src/coreclr/jit/importer.cpp

```diff
@@ -13931,3 +13937,121 @@ methodPointerInfo* Compiler::impAllocateMethodPointerInfo(const CORINFO_RESOLVED
+        divNode->gtFlags |= GTF_UNSIGNED;
+
+    divNode = gtFoldExpr(divNode);
+
+    // Is is still a division after folding? If not - push the result and finish.
+    if (!divNode->OperIs(GT_DIV, GT_UDIV, GT_MOD, GT_UMOD))
+    {
+        impPopStack();
+        impPopStack();
+        impPushOnStack(divNode, tiRetVal);
+        return true;
+    }
+
+    if (divisor->IsNeverZero() && divisor->IsNeverNegativeOne(this))
+    {
+        divNode->gtFlags |= (GTF_DIV_MOD_NO_OVERFLOW | GTF_DIV_MOD_NO_BY_ZERO);
+        assert(!divNode->OperMayThrow(this));
+        impPopStack();
+        impPopStack();
+        impPushOnStack(divNode, tiRetVal);
+        return true;
+    }
+
+    GenTree* result = divNode;
+
+    impPopStack();
+    impPopStack();
+    impSpillSideEffects(false, CHECK_SPILL_ALL DEBUGARG("spilling side effects before adding runtime exception"));
+
+    // The dividend and divisor must be cloned in this order to preserve evaluation order.
+    unsigned dividendTmp = lvaGrabTemp(true DEBUGARG("dividend used in runtime checks"));
+    impStoreToTemp(dividendTmp, dividend, CHECK_SPILL_NONE);
+    dividend               = gtNewLclvNode(dividendTmp, genActualType(dividend));
+    divNode->AsOp()->gtOp1 = dividend;
+
+    unsigned divisorTmp = lvaGrabTemp(true DEBUGARG("divisor used in runtime checks"));
+    impStoreToTemp(divisorTmp, divisor, CHECK_SPILL_NONE);
+    divisor                = gtNewLclvNode(divisorTmp, genActualType(divisor));
+    divNode->AsOp()->gtOp2 = divisor;
+
+    // Expand division into QMark containing runtime checks.
+    if (!isUnsigned)
+    {
+        assert(!(varTypeIsUnsigned(dividend) || varTypeIsUnsigned(divisor)));
+
+        dividend = gtClone(dividend, false);
+        divisor  = gtClone(divisor, false);
+
+        const ssize_t minValue = genActualType(dividend) == TYP_LONG ? INT64_MIN : INT32_MIN;
+
+        // (dividend == MinValue && divisor == -1)
+        GenTreeOp* const divisorIsMinusOne =
+            gtNewOperNode(GT_EQ, TYP_INT, divisor, gtNewIconNode(-1, genActualType(divisor)));
+        GenTreeOp* const dividendIsMinValue =
+            gtNewOperNode(GT_EQ, TYP_INT, dividend, gtNewIconNode(minValue, genActualType(dividend)));
+
+        GenTreeOp* const combinedTest = gtNewOperNode(GT_AND, TYP_INT, divisorIsMinusOne, dividendIsMinValue);
+        GenTree*         condition    = gtNewOperNode(GT_EQ, TYP_INT, combinedTest, gtNewTrue());
+
+        result =
+            gtNewQmarkNode(resultType, condition,
+                           gtNewColonNode(resultType, gtNewHelperCallNode(CORINFO_HELP_OVERFLOW, resultType), result));
+    }
+
+    divisor = gtClone(divisor, false);
+    result =
+        gtNewQmarkNode(resultType,
+                       // (divisor == 0)
+                       gtNewOperNode(GT_EQ, TYP_INT, divisor, gtNewIconNode(0, genActualType(divisor))),
+                       gtNewColonNode(resultType, gtNewHelperCallNode(CORINFO_HELP_THROWDIVZERO, resultType), result));
```

Comment:
`gtThenLikelihood`  should be set for these qmark nodes to indicate that we never expect the exceptions to be thrown. I do not think the current perfscore diffs show the right picture because this is missing.